### PR TITLE
upsample nearest & bilinear 2d

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

File: src/frontends/pytorch/src/op/upsample.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if (!context.input_is_none(2)) {
+            align_corners = context.const_input<bool>(2);
+        }
+    }
+    auto target_axes = std::make_shared<opset8::Constant>(element::i32, Shape{2}, std::vector<int>({2, 3}));
+    auto scales = context.mark_node(std::make_shared<opset8::Constant>(element::f32, Shape{2}, std::vector<double>({1, 1})));
+    auto output_sizes = std::make_shared<opset8::Constant>(element::i32, Shape{2}, std::vector<int>({1, 1}));
+    if (context.input_is_none(1)) {
+        FRONT_END_OP_CONVERSION_CHECK(!context.input_is_none(scale_id), "Scale or Output size should be provided");
+        auto spatial_scales = context.const_input<std::vector<double>>(scale_id);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/upsample.cpp]: I think `scales` can be non-constant in many cases, so better would be to have sub-graph for supporting scales (with If operation I guess)

File: src/frontends/pytorch/src/op/upsample.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        if (spatial_scales.size() == 1) {
+            spatial_scales.push_back(spatial_scales[0]);
+        }
+        size_mode = opset8::Interpolate::ShapeCalcMode::SCALES;
+        scales = std::make_shared<opset8::Constant>(element::f32, Shape{spatial_scales.size()}, spatial_scales);
+    } else {
+        auto out_sizes = context.const_input<std::vector<int64_t>>(1);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/upsample.cpp]: Same as for `scales`, better to support non-constant case.

**Comment by mvafin** [comment]
nit: it would be better to have a single file for layer tests of both upsamples, since the tests are very similar.

File: src/core/src/op/interpolate.cpp [hunk 3]
```diff
@@ -202,7 +202,7 @@ void op::v4::Interpolate::validate_and_infer_types() {
     NODE_VALIDATION_CHECK(this,
                           input_et == element::f32 || input_et == element::f16 || input_et == element::i8 ||
                               input_et == element::bf16 || input_et == element::u8 || input_et == element::i64 ||
-                              input_et == element::i32,
+                              input_et == element::i32 || input_et == element::dynamic,
```
Comment (slyalin) [inline src/core/src/op/interpolate.cpp:205]: Don't need it. Dynamic means any time. Dynamic should be passed through.

File: src/frontends/pytorch/src/op/upsample.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/upsample.cpp]: handelered (but I find simplier solution then if, just multipy to const of required size)

File: src/frontends/pytorch/src/op/upsample.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/upsample.cpp]: updated

**Comment by eaidova** [comment]
> nit: it would be better to have a single file for layer tests of both upsamples, since the tests are very similar.

@mvafin I refactored tests, now they both tested in one test class

**Review by mvafin — APPROVED** [review]
Looks good to me

**Pull request merged**